port/unix/omrsysinfo.c

```diff
@@ -6121,14 +6137,37 @@ getCgroupMemoryLimit(struct OMRPortLibrary *portLibrary, uint64_t *limit)
 {
 	uint64_t cgroupMemLimit = 0;
 	uint64_t physicalMemLimit = 0;
-	int32_t numItemsToRead = 1; /* memory.limit_in_bytes file contains only one integer value */
+	int32_t numItemsToRead = 1; /* memory.limit_in_bytes (v1) and memory.max (v2) files contains only one integer value */
 	int32_t rc = 0;
 
 	Trc_PRT_sysinfo_cgroup_get_memlimit_Entry();
 
-	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, "memory.limit_in_bytes", numItemsToRead, "%" SCNu64, &cgroupMemLimit);
-	if (0 != rc) {
-		Trc_PRT_sysinfo_cgroup_get_memlimit_memory_limit_read_failed("memory.limit_in_bytes", rc);
+	if (isCgroupV1Available(portLibrary)) {
+		rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, "memory.limit_in_bytes", numItemsToRead, "%" SCNu64, &cgroupMemLimit);
+		if (0 != rc) {
+			Trc_PRT_sysinfo_cgroup_get_memlimit_memory_limit_read_failed("memory.limit_in_bytes", rc);
+			goto _end;
+		}
+	} else if (isCgroupV2Available(portLibrary)) {
+		/* Cgroup v2 memory.max file contains a number/string in [0, max], where "max" means no limit */
+		char memLimitString[20];
+		rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, "memory.max", numItemsToRead, "%s", memLimitString);
+		if (0 != rc) {
+			Trc_PRT_sysinfo_cgroup_get_memlimit_memory_limit_read_failed("memory.max", rc);
+			goto _end;
+		}
+		if (0 == strcmp(memLimitString, "max")) {
+			cgroupMemLimit = UINT64_MAX;
+		} else {
+			rc = sscanf(memLimitString, "%" SCNu64, &cgroupMemLimit);
```

Comment:
Yes according to doc all memory for cgroups is set in bytes. getPhysicalMemory is too since it's multiplying page size with number of pages (both in bytes)